frontend/server/src/main/java/org/pytorch/serve/wlm/WorkLoadManager.java

```diff
@@ -109,6 +120,17 @@ public int getNumRunningWorkers(ModelVersionName modelVersionName) {
             }
 
             int currentWorkers = threads.size();
+            boolean isRestartWorkers = isLauncherRestartWorkers(currentWorkers);
+
+            if (isRestartWorkers) {
+                logger.warn(
+                        "removing {} current thread(s) prior to restarting {} thread(s)",
+                        currentWorkers,
+                        minWorker);
+                maxWorker = 0;
+                minWorker = 0;
```

Comment:
By setting `minWorker=0` when `currentWorkers > 0`, this will never enter the 'if (currentWorkers < minWorker)' condition in line 134. And by setting `maxWorker=0`  when `currentWorkers > 0`, this will always enter the `else` condition in line 136. The  `for` loop in line 115 will proceed to remove all  `currentWorkers` as `maxWorker=0`.